template.go

```diff
@@ -7,44 +7,35 @@ import (
 	"github.com/Masterminds/sprig"
 	"io/ioutil"
 	"log"
+	"os"
 	"path/filepath"
 	"text/template"
 )
 
-type OptionalString struct {
-	ptr *string
-}
-
-func (s OptionalString) String() string {
-	if s.ptr == nil {
-		return ""
+func Require(arg string) (string, error) {
+	fmt.Fprintf(os.Stderr, "WARNING: require built-in function will be deprecated in future release. Use required instead.\n")
+	if len(arg) == 0 {
+		return "", errors.New("Required argument is missing or empty!")
 	}
-	return *s.ptr
+	return arg, nil
 }
 
-func Require(arg interface{}) (string, error) {
-	if arg == nil {
-		return "", errors.New("Required argument is missing!")
-	}
-
-	switch v := arg.(type) {
-	case string:
-		return v, nil
-	case *string:
-		if v != nil {
-			return *v, nil
-		}
-	case OptionalString:
-		if v.ptr != nil {
-			return *v.ptr, nil
+// copied from Helm source:
+// https://github.com/kubernetes/helm/blob/78d6b930bd325ed87b297c57b02fc7c9c7dfcfac/pkg/engine/engine.go#L156-L165
+func Required(warn string, val interface{}) (interface{}, error) {
```

Comment:
Since val is of type interface{} would it be useful to cover other types than string in tests?

Reply:
If you copy code form other repository you have to copy license file too. I suggest to put is to `3th-party-licences/github.com/kubernetes/helm/LICENSE` file.

Reply:
Helm is under Apache v2 lincence, goenvtemplator is GPL v3, if I understand it correctly, it is apache licence is permissive and compatible with GPL v3.

https://en.wikipedia.org/wiki/Apache_License

Reply:
If you have nothing to add @lksv I consider this resolved.